File: backend/utils/populationdata/quiz_answer.py
from typing import List
from pydantic import Json

quizAnswers: List[Json] = [
    {
        "student_id": 3,
        "quiz_id": 11,
        "marks_obtained": 20,
        "options_selected": {
            "25": 1,
            "26": 2,
            "27": 0,
            "28": [0, 2, 3],
        },
    },
    {
        "student_id": 2,
        "quiz_id": 11,
        "marks_obtained": 20,
        "options_selected": {
            "25": 1,
            "26": 0,
            "27": 2,
            "28": [0, 2],
        },
    },
    {
        "student_id": 4,
        "quiz_id": 11,
        "marks_obtained": 12,
        "options_selected": {
            "25": 1,
            "26": 0,
            "27": 2,
            "28": [0, 2],
        },
    },
    {
        "student_id": 3,
        "quiz_id": 1,
        "marks_obtained": 32,
        "options_selected": {
            "1": [2, 3],
            "6": 1,
            "15": [1, 4],
            "18": 3,
        },
    },
]

File: mobile/lib/repository/models/quiz_answer.dart
import 'package:equatable/equatable.dart';
import 'package:json_annotation/json_annotation.dart';

part 'quiz_answer.g.dart';

@JsonSerializable()
class QuizAnswer extends Equatable {
  final int? id;
  final int? student_id;
  final int? quiz_id;
  final int? marks_obtained;
  final Map? options_selected;

  const QuizAnswer({
    this.id,
    this.student_id,
    this.quiz_id,
    this.marks_obtained,
    this.options_selected,
  });

  static const empty = QuizAnswer(
      id: null,
      student_id: null,
      quiz_id: null,
      marks_obtained: null,
      options_selected: null);

  factory QuizAnswer.fromJson(Map<String, dynamic> json) =>
      _$QuizAnswerFromJson(json);

  Map<String, dynamic> toJson() => _$QuizAnswerToJson(this);

  @override
  List<Object?> get props =>
      [id, student_id, quiz_id, marks_obtained, options_selected];
}
